unit/solvers/smt2_incremental/smt_sorts.cpp

```diff
@@ -0,0 +1,32 @@
+// Author: Diffblue Ltd.
+
+#include <testing-utils/use_catch.h>
+
+#include <solvers/smt2_incremental/smt_sorts.h>
+
+TEST_CASE("Test upcasts of smt_bool_sortt.", "[core][smt2_incremental]")
+{
+  const smt_bool_sortt bool_sort;
+  smt_sortt sort{bool_sort};
+  sort = bool_sort;
+}
+
+TEST_CASE("Test upcasts of smt_bit_vec_sortt.", "[core][smt2_incremental]")
+{
+  const smt_bit_vector_sortt bit_vec_sort{32};
+  smt_sortt sort{bit_vec_sort};
+  sort = bit_vec_sort;
```

Comment:
It tests that the copy constructor of the `smt_sortt` class can be called with a smt_bit_vector_sortt typed parameter and the same for the assignment operator. Effectively testing the upcast which is carried out in these two operations is accessible. I added these two tests because the same test with the `irept` constructor should fail due to the access restrictions. Writing a test that it does not work with `irept`, could be done using some template meta programming to write a "does not compile" function, but probably would not be straight forward code to for us to read or write.